src/shared/GitHub.UI/Controls/SixDigitInput.axaml.cs

```diff
@@ -0,0 +1,194 @@
+            {
+                SetAndRaise(TextProperty, ref _text, value);
+                if (!_ignoreTextBoxUpdate) SetTextBoxes(value);
+            }
+        }
+
+        private void SetTextBoxes(string text)
+        {
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                foreach (TextBox textBox in _textBoxes)
+                {
+                    textBox.Text = string.Empty;
+                }
+            }
+            else
+            {
+                IEnumerable<char> digits = text.Where(char.IsDigit);
+                string digitsStr = string.Join(string.Empty, digits).PadRight(6);
+                for (int i = 0; i < digitsStr.Length; i++)
+                {
+                    _textBoxes[i].Text = digitsStr.Substring(i, 1);
+                }
+            }
+        }
+
+        public void SetFocus()
+        {
+            KeyboardDevice.Instance.SetFocusedElement(_textBoxes[0], NavigationMethod.Tab, KeyModifiers.None);
+        }
+
+        private void SetUpTextBox(TextBox textBox)
+        {
+            textBox.AddHandler(KeyDownEvent, OnPreviewKeyDown, RoutingStrategies.Tunnel);
+
+            void OnPreviewKeyDown(object sender, KeyEventArgs e)
+            {
+                // Handle paste
+                if (_keyMap.Paste.Any(x => x.Matches(e)))
+                {
+                    OnPaste();
+                    e.Handled = true;
+                }
+                // Handle keyboard navigation
+                else if (e.Key == Key.Left || e.Key == Key.Right || e.Key == Key.Back)
+                {
+                    e.Handled = e.Key == Key.Right ? MoveNext() : MovePrevious();
+                    if (e.Key == Key.Back)
+                    {
+                        textBox.Text = string.Empty;
+                    }
+                }
+                // Only allow 0-9, Tab, Escape, and Delete
+                else if (e.Key != Key.D0 &&
+                         e.Key != Key.D1 &&
+                         e.Key != Key.D2 &&
+                         e.Key != Key.D3 &&
+                         e.Key != Key.D4 &&
+                         e.Key != Key.D5 &&
+                         e.Key != Key.D6 &&
+                         e.Key != Key.D7 &&
+                         e.Key != Key.D8 &&
+                         e.Key != Key.D9 &&
+                         e.Key != Key.NumPad0 &&
+                         e.Key != Key.NumPad1 &&
+                         e.Key != Key.NumPad2 &&
+                         e.Key != Key.NumPad3 &&
+                         e.Key != Key.NumPad4 &&
+                         e.Key != Key.NumPad5 &&
+                         e.Key != Key.NumPad6 &&
+                         e.Key != Key.NumPad7 &&
+                         e.Key != Key.NumPad8 &&
+                         e.Key != Key.NumPad9 &&
+                         e.Key != Key.Tab &&
+                         e.Key != Key.Escape &&
+                         e.Key != Key.Delete)
```

Comment:
At what point is it better to have a static `HashSet` and check `.Contains()`?

Reply:
I'm not sure.. right now I was just copying what was done in the WPF-version of this custom control.